vulcanai/datasets/tabular_data_utils.py

```diff
@@ -9,12 +9,42 @@
 import logging
 from itertools import groupby
 from sklearn import preprocessing
+import torch
+from torch.utils.data import TensorDataset
 
 logger = logging.getLogger(__name__)
-
 # TODO: split function
 
 
+def convert_to_tensor_datasets(df, target_vars=None):
+    """
+    Given a df, returns a TensorDataset, with the target variables contained
+    in the second tensor being specified by the column name(s) in target_vars.
+    Args:
+        df: Dataframe
+            The dataframe to be operated on
+        target_vars: string or list of string
+            The column(s) to be used in the target tensor.
+
+    Returns: TensorDataset
+        The resulting dataset representing the data contained in the df
+
+    """
+
+    if not target_vars:
+        return TensorDataset(torch.Tensor(np.array(df)))
+
+    if target_vars and not target_vars.isinstance(list):
+        target_vars = [target_vars]
+
+    data = torch.Tensor(np.array(df.drop(target_vars, axis=1)))
+    target = torch.LongTensor(np.array(df[target_vars]))
+
+    dataset = TensorDataset(data, target)
```

Comment:
did you want to return multiple datasets for the data and all the targets? or do you think it's fine to send it as one?

Reply:
Also what if the target isn't a class but rather a float value? perhaps we should infer the type from the df for the targets?

Reply:
that's how we need it to be for training and inference in vulcan

good point about the float values, I'll fix that.